agent/structs/config_entry_oss_test.go

```diff
@@ -0,0 +1,101 @@
+// +build !consulent
+
+package structs
+
+import (
+	"github.com/hashicorp/hcl"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestDecodeConfigEntry_OSS(t *testing.T) {
+
+	for _, tc := range []struct {
+		name      string
+		camel     string
+		snake     string
+		expect    ConfigEntry
+		expectErr string
+	}{
+		{
+			name: "namespaces invalid top level",
+			snake: `
+				kind = "terminating-gateway"
+				name = "terminating-gateway"
+				namespace = "foo"
+			`,
+			camel: `
+				Kind = "terminating-gateway"
+				Name = "terminating-gateway"
+				Namespace = "foo"
+			`,
+			expectErr: `invalid config key "namespace", namespaces are a consul enterprise feature`,
+		},
+		{
+			name: "namespaces invalid deep",
+			snake: `
+				kind = "ingress-gateway"
+				name = "ingress-web"
+				listeners = [
+					{
+						port = 8080
+						protocol = "http"
+						services = [
+							{
+								name = "web"
+								hosts = ["test.example.com", "test2.example.com"]
+								namespace = "frontend"
+							},
+						]
+					}
+				]
+			`,
+			camel: `
+				Kind = "ingress-gateway"
+				Name = "ingress-web"
+				Namespace = "blah"
+				Listeners = [
+					{
+						Port = 8080
+						Protocol = "http"
+						Services = [
+							{
+								Name = "web"
+								Hosts = ["test.example.com", "test2.example.com"]
+								Namespace = "frontend"
+							},
+						]
+					},
+				]
+			`,
+			expectErr: `* invalid config key "listeners[0].services[0].namespace", namespaces are a consul enterprise feature`,
+		},
+	} {
+		tc := tc
+
+		testbody := func(t *testing.T, body string) {
+			t.Helper()
```

Comment:
Minor thing (and I see this was copied from the existing test), but I think using `t.Helper()` here is not helpful (please forgive the pun).

With `t.Helper()` the line number that gets reported is the one line in the `t.Run()` below (line 95, or 98). Since there is only a single line in the t.Run() body, the line number doesn't give us any more information than the name of the test.

Without the `t.Helper()` we would get a line number in the `testbody` function, which gives us more information, and we still have the test name to identify the test case.

The distinction being "test helper function" vs "test case function". Only helpers would get `t.Helper`, and this func seems to be an implementation of a test case.

Reply:
I see what you mean. Getting the line in `testbody` does seem more informative. I updated that.